src/dataflow-types/src/client/replicated.rs

```diff
@@ -213,11 +214,14 @@ where
                     self.peeks.remove(uuid);
                 }
                 // Enqueue the response to the cancelation.
-                self.pending_response.extend(
-                    uuids
-                        .iter()
-                        .map(|uuid| ComputeResponse::PeekResponse(*uuid, PeekResponse::Canceled)),
-                );
+                self.pending_response.extend(uuids.iter().map(|uuid| {
+                    // TODO(guswynn): populate OpenTelemetryContext::empty()
+                    ComputeResponse::PeekResponse(
+                        *uuid,
+                        PeekResponse::Canceled,
+                        OpenTelemetryContext::empty(),
+                    )
+                }));
```

Comment:
I would change `peeks` to store the `otel_context` when the `ComputeCommand::Peek` arrives. Then you can pull it out of `self.peeks` here and use it to populate the cancellation response.